TempoCore/Content/Python/API/tempo/_tempo_context.py

```diff
@@ -1,32 +1,53 @@
 # Copyright Tempo Simulation, LLC. All Rights Reserved
 
+import asyncio
 import grpc
 import os
 import sys
 
 
-class TempoContext(object):
-    _instance = None
+def get_event_loop():
+    try:
+        loop = asyncio.get_event_loop()
+    except RuntimeError:
+        loop = asyncio.new_event_loop()
+        asyncio.set_event_loop(loop)
+    return loop
+
 
-    def __new__(cls):
-        if not cls._instance:
-            cls._instance = super(TempoContext, cls).__new__(cls)
-        return cls._instance
+def set_server(address="localhost", port=10001):
+    loop = get_event_loop()
+    loop.run_until_complete(tempo_context()._set_server(address, port))
 
+
+class TempoContext(object):
     def __init__(self):
-        self.set_server_address("localhost:10001")
+        print("__init__: {}".format(id(self)))
+        self.server_address = "localhost"
+        self.server_port = "10001"
+        self._init_channel()
         self.stubs = {}
-        self.message_classes = {}
 
     def get_stub(self, stub_class):
         return self.stubs.setdefault(stub_class, stub_class(self.channel))
 
-    def set_server_address(self, server_address):
-        self.channel = grpc.aio.insecure_channel(server_address,
-            options=[
-                ('grpc.max_receive_message_length', 1000000000), # 1Gb
-            ]
-        )
+    async def _set_server(self, address="localhost", port="10001"):
+        self.server_address = address
+        self.server_port = port
+        self._init_channel()
+
+    def _init_channel(self):
+        self.channel = grpc.aio.insecure_channel("{}:{}".format(self.server_address, self.server_port),
+                                                 options=[
+                                                     ('grpc.max_receive_message_length', 1000000000), # 1Gb
+                                                 ]
+                                                 )
+        self.stubs = {}
+
+
+# "Singleton Factory" https://stackoverflow.com/questions/12305142/issue-with-singleton-python-call-two-times-init
```

Comment:
Our previous singleton used only one instance of the object but still had `__init__` called every time it was accessed. This resolves that issue.